host/install.sh

```diff
@@ -0,0 +1,170 @@
+{
+    ECHO "${YELLOW_BOLD}Warning: ${YELLOW}$*${NO_COLOR}" > /dev/stderr
+}
+
+function ERROR()
+{
+    ECHO "${RED_BOLD}ERROR: ${RED}$*${NO_COLOR}"
+    exit 1
+}
+
+function install_pip_requirements()
+{
+    local requirements_file="$1"
+
+    if [ ! -f ${requirements_file} ]; then
+        WARNING "No ${requirements_file} found"
+        return  # Exit the function if the file exists
+    fi
+
+    INFO "install_pip_requirements( ${requirements_file} ):"
+    if pip install -r "$requirements_file" >/dev/null 2>&1; then
+        INFO "pip install completed"
+    else
+        ERROR "process_and_update_sha(): pip install failed."
+        return 1  # Exit the function with error code
+    fi
+}
+
+function clone_repo()
+{
+    # Requirment it to clone only if not present.
+    local repo_url="$1"
+    local path="$2"
+    local version="$3"
+    local message="$4"
+
+    if [[ -z "${repo_url}" ]]; then
+        ERROR "clone_repo:A url for a repository must be passed to the clone repo function"
+    fi
+    if [[ -z "${version}" ]]; then
+        ERROR "clone_repo:Version not specified"
+    fi
+    if [[ ! -z "${path}" ]]; then
+        if [[ ! -d "${path}" ]]; then
+            INFO "git clone ${repo_url} @ ${version} ${CYAN}${message}${NO_COLOR}"
+            git clone ${repo_url} "${path}" > /dev/null 2>&1
+            cd ${path}
+            #INFO "git checkout ${version}"
+            git checkout ${version} > /dev/null 2>&1
+            cd - > /dev/null
+        fi
+    fi
+}
+
+function setup_and_enable_venv()
+{
+    # Check if virtual environment directory exists, create if not
+    if [[ ! -d "$VENV_DIR" ]]; then
+        ECHO "Creating Virtual environment ${YELLOW}'$VENV_NAME'${NO_COLOR}"
+        python3 -m venv "$VENV_DIR"
+        ECHO "Virtual environment created."
+    fi
+
+    # Request that the user re-run this script from the vendor
+    # Check if already inside a virtual environment
+    if [[ ! -n "$VIRTUAL_ENV" ]]; then
+        # Activate virtual environment
+        ECHO "please source & re-run ${YELLOW}install.sh${NO_COLOR} to ensure setup:"
+        ECHO ${YELLOW}"source ./activate_venv.sh"${NO_COLOR}
+        exit 1  # Exit the function if already in a venv
+    fi
+
+    if [ -f "${VENV_DIR}/.installed" ]; then
+        return
+    fi
+
+    # Upgrade pip
+    #python3 -m pip install --upgrade pip
+    #echo "pip upgraded within the virtual environment."
+
+    touch ${VENV_DIR}/.installed
+}
+
+
+## Setup and start venv
+setup_and_enable_venv
+
+### Clone required repos ###
+# Setup raft
+clone_repo git@github.com:rdkcentral/python_raft.git "${RAFT_DIR}" "1.1.1" "in ./raft"
+install_pip_requirements "${RAFT_DIR}"/requirements.txt
+
+# Setup ut-raft
+clone_repo git@github.com:rdkcentral/ut-raft.git "${PLUGINS_FRAMEWORK_DIR}/ut_raft" 2.0.0 "in ./raft/framework/plugins"
+install_pip_requirements "${PLUGINS_FRAMEWORK_DIR}"/ut_raft/requirements.txt
+
+## Install your own sub git repo's in here as required
+#
```

Comment:
create ut/host/requirements.txt with librosa
update install.sh to downlaod the required package
install_pip_requirements "${PWD}"/requirements.txt